api/plugin.go

```diff
@@ -455,3 +455,60 @@ func (s *Server) initializePlugin(pluginType string) (plugin.Plugin, error) {
 		return nil, fmt.Errorf("unknown plugin type: %s", pluginType)
 	}
 }
+
+func (s *Server) GetPlugins(c echo.Context) error {
+	plugins, err := s.db.FindPlugins(c.Request().Context())
+	if err != nil {
+		message := map[string]interface{}{
+			"message": "failed to get plugins",
+		}
+		s.logger.Error(err)
+		return c.JSON(http.StatusInternalServerError, message)
+	}
+
+	return c.JSON(http.StatusOK, plugins)
+}
+
+func (s *Server) GetPlugin(c echo.Context) error {
+	pluginID := c.Param("pluginId")
+	if pluginID == "" {
+		err := fmt.Errorf("plugin id is required")
+		message := map[string]interface{}{
+			"message": "failed to get plugin",
+			"error":   err.Error(),
+		}
+		s.logger.Error(err)
+
+		return c.JSON(http.StatusBadRequest, message)
+	}
+
+	plugin, err := s.db.FindPluginById(c.Request().Context(), pluginID)
+	if err != nil {
+		message := map[string]interface{}{
+			"message": "failed to get plugin",
+		}
+		s.logger.Error(err)
+		return c.JSON(http.StatusInternalServerError, message)
+	}
+
+	return c.JSON(http.StatusOK, plugin)
+}
+
+func (s *Server) CreatePlugin(c echo.Context) error {
+	// TODO: this parses json, but does not validate it (required fields)
```

Comment:
yes, we need to implement validations separately

Reply:
Added a common validator [here](https://github.com/LimeChain/vultiserver-plugin/pull/40/commits/e85bde086d12a8ee9c905ba20567fd5c1e4a451a)

tbh, the default error message looks a bit cryptic, but it may be ok for now

```
{
    "message": "Key: 'PluginCreateDto.ServerEndpoint' Error:Field validation for 'ServerEndpoint' failed on the 'required' tag"
}
```